ui_tests/pages/profiles_page.py

```diff
@@ -0,0 +1,299 @@
+            "walk": "//button[@data-test='sc4snmp:form:condition-walk']",
+            "conditional": "//button[@data-test='sc4snmp:form:condition-conditional']",
+        }
+        profile_type_expander_xpath = (
+            "//button[@data-test='sc4snmp:form:select-condition']"
+        )
+        expander = driver.find_element(By.XPATH, profile_type_expander_xpath)
+        expander.click()
+        option = driver.find_element(By.XPATH, profiles[profile_type])
+        option.click()
+
+    def set_frequency(self, freq_value):
+        logger.info(f"Setting profile frequency: {freq_value}")
+        xpath = "//div[@data-test='sc4snmp:form:frequency-input']//span//input"
+        freq_field = driver.find_element(By.XPATH, xpath)
+        helper.clear_input(freq_field)
+        # freq_field.send_keys(Keys.BACKSPACE)  # clear() is not working here
+        freq_field.send_keys(freq_value)
+
+    def set_profile_name(self, name):
+        logger.info(f"Setting profile frequency: {name}")
+        xpath = "//div[@data-test='sc4snmp:form:profile-name-input']//span//input"
+        name_input = driver.find_element(By.XPATH, xpath)
+        helper.clear_input(name_input)  # this is useful when editing profile name
+        name_input.send_keys(name)
+
+    def add_varBind(self, mcomponent, mobject=None, mindex=None):
+        logger.info(f"Adding varBind: {mcomponent, mobject, mindex}")
+        add_varBind_button_xpath = "//div[@data-test='sc4snmp:form:add-varbinds']//span[contains(text(),'Add varBind')]"
+        add_varBind_btn = driver.find_element(By.XPATH, add_varBind_button_xpath)
+        add_varBind_btn.click()
+        varbind_row_xpath = "//div[@data-test='sc4snmp:form:varbind-row']"
+        varBinds_rows = driver.find_elements(By.XPATH, varbind_row_xpath)
+        component_xpath = (
+            "//div[@data-test='sc4snmp:form:varbind-mib-component-input']/span/input"
+        )
+        component_input = varBinds_rows[-1].find_element(By.XPATH, component_xpath)
+        component_input.send_keys(mcomponent)
+        if mobject is not None:
+            object_xpath = (
+                "//div[@data-test='sc4snmp:form:varbind-mib-object-input']/span/input"
+            )
+            object_input = varBinds_rows[-1].find_element(By.XPATH, object_xpath)
+            object_input.send_keys(mobject)
+        if mindex is not None:
+            index_xpath = (
+                "//div[@data-test='sc4snmp:form:varbind-mib-index-input']/span/input"
+            )
+            index_input = varBinds_rows[-1].find_element(By.XPATH, index_xpath)
+            index_input.send_keys(mindex)
+
+    def edit_varBind(self, new_mcomponent, new_mobject, new_mindex):
+        logger.info(
+            f"Editing varBind new values: {new_mcomponent}, {new_mobject}, {new_mindex}"
+        )
+        varbind_row_xpath = "//div[@data-test='sc4snmp:form:varbind-row']"
+        varBinds_row = driver.find_element(By.XPATH, varbind_row_xpath)
```

Comment:
In `edit_varBind` do we expect that only one row with a varbind exists? In `add_varBind` I noticed that this variable is called `varBinds_rows` and later we extract each element of varbind row like that `varBinds_rows[-1].find_element(...)`, meanwhile here we access the using ` varBinds_row.find_element(...)`

Reply:
yes, this method edits first varbind row and it is intentional as this method is used only once in `test_edit_profile`